vcert/connection_cloud.py

```diff
@@ -167,69 +190,82 @@ def _parse_policy_responce_to_object(d):
         return policy
 
     def _get_policy_by_id(self, policy_id):
-        status, data = self._get(URLS.TEMPLATE_BY_ID % policy_id)
+        app_name, cit_alias = _parse_zone(policy_id)
+        status, data = self._get(URLS.CERTIFICATE_TEMPLATE_BY_ID % (app_name, cit_alias))
         if status != HTTPStatus.OK:
-            log.error("Invalid status during geting policy: %s for policy %s" % (status, policy_id))
+            log.error("Invalid status during getting policy: %s for policy %s" % (status, policy_id))
             raise ServerUnexptedBehavior
-        return self._parse_policy_responce_to_object(data)
+        return self._parse_policy_response_to_object(data)
 
     def auth(self):
         status, data = self._get(URLS.USER_ACCOUNTS)
         if status == HTTPStatus.OK:
             return data
 
-    def _get_zone_id_by_tag(self, tag):
+    def _get_app_details_by_name(self, app_name):
         """
-        :param str tag:
-        :rtype Zone
+        :param str app_name:
+        :rtype AppDetails
         """
-        if not tag:
-            raise ClientBadData("You need to specify zone tag")
-        status, data = self._get(URLS.ZONE_BY_TAG % tag)
+        if not app_name:
+            raise ClientBadData("You need to specify the application name")
+        status, data = self._get(URLS.APP_DETAILS_BY_NAME % app_name)
         if status == HTTPStatus.OK:
-            return data['id']
+            return AppDetails(data["id"], data["certificateIssuingTemplateAliasIdMap"])
         elif status in (HTTPStatus.BAD_REQUEST, HTTPStatus.NOT_FOUND, HTTPStatus.PRECONDITION_FAILED):
             log_errors(data)
         else:
             pass
 
     def request_cert(self, request, zone):
-        zone_id = self._get_zone_id_by_tag(zone)
+        app_name, cit_alias = _parse_zone(zone)
+        details = self._get_app_details_by_name(app_name)
+        cit_alias_decoded = urlparse.unquote(cit_alias)
+        cit_id = details.cit_alias_id_map.get(cit_alias_decoded)
         if not request.csr:
             request.build_csr()
-        status, data = self._post(URLS.CERTIFICATE_REQUESTS,
-                                  data={"certificateSigningRequest": request.csr, "zoneId": zone_id})
+
+        ip_address = get_ip_address()
+        status, data = self._post(URLS.CERTIFICATE_REQUESTS, data={
+            "certificateSigningRequest": request.csr,
+            "applicationId": details.app_id,
+            "certificateIssuingTemplateId": cit_id,
+            # TODO remove validity period workaround
+            "validityPeriod": "PT48H",
```

Comment:
This can be removed now that Venafi Cloud bug requiring `valdityPeriod` has been fixed and VCert-Python doesn't yet support the flexible validity period feature.

Reply:
Workaround has been removed.